File: myapp/helpers/quickField.js
module.exports = function (fieldType, fieldName, label, width, fieldValue, sqlOrClass, columnToShow, additionalTags) {
    this.buildField = function () {
        if (typeof fieldType === "string") {
            for (let f in this.fieldsModel) {
                if (this.fieldsModel[f].name === fieldType) {
                    fieldType = f;
                }
            }
        }

        let field = this.getFieldByType(fieldType);

        this.field = setCol(field, width);
    };

    this.fieldsModel = {
        1: {
            name: "text",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
        2: {
            name: "memo",
            getHtml: function () {
                return getLabel() + "<textarea class='form-control'></textarea>";
            }
        },
        3: {
            name: "simpleSelect",
            getHtml: function () {
                return getLabel() + getSelect();
            }
        },
        4: {
            name: "Radio/grava texto",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
        5: {
            name: "Checkboxes/grava texto",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
        6: {
            name: "currency",
            getHtml: function () {
                return getLabel() + getInput("currency");
            }
        },
        7: {
            name: "number",
            getHtml: function () {
                return getLabel() + getInput(false, "number");
            }
        },
        8: {
            name: "CPF",
            getHtml: function () {
                return getLabel() + getInput("cpf");
            }
        },
        9: {
            name: "CNPJ",
            getHtml: function () {
                return getLabel() + getInput("cnpj");
            }
        },
        10: {
            name: "datepicker",
            getHtml: function () {
                return getLabel() + getInput("datepicker");
            }
        },
        11: {
            name: "Hora",
            getHtml: function () {
                return getLabel() + getInput("time");
            }
        },
        12: {
            name: "Data e Hora",
            getHtml: function () {
                return getLabel() + getInput("datetime");
            }
        },
        13: {
            name: "Data Máscara",
            getHtml: function () {
                return getLabel() + getInput("date");
            }
        },
        14: {
            name: "phone",
            getHtml: function () {
                return getLabel() + getInput("phone");
            }
        },
        15: {
            name: "mobile",
            getHtml: function () {
                return getLabel() + getInput("phone");
            }
        },
        16: {
            name: "Cep",
            getHtml: function () {
                return getLabel() + getInput("cep");
            }
        },
        17: {
            name: "Imagem Única",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
        18: {
            name: "Múltiplo com Termos/grava texto",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
        19: {
            name: "Colorpicker Simples",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
        20: {
            name: "Colorpicker Médio",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
        21: {
            name: "editor",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
        22: {
            name: "email",
            getHtml: function () {
                return getLabel() + getInput(false, "email");
            }
        },
        23: {
            name: "contratado",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
        24: {
            name: "empresa",
            getHtml: function () {
                return getLabel() + getInput();
            }
        },
    };

    const setCol = function (field, size) {
        return "<div class='col-md-" + size + "'>" + field + "</div>";
    };

    function getInput(classes, type) {
        const isRequired = false;
        // const isRequired = obj.required === 1;
        const typeVal = typeof type !== "undefined" ? type : "text";
        return "<input type='" + typeVal + "' " + (isRequired ? "required" : "") + " " +
            "class='form-control " + (typeof classes !== "undefined" && classes ? classes : "") + "' name='" + fieldName + "'>";
    }

    function getSelect() {
        const isRequired = false;
        // const isRequired = obj.required === 1;

        return "<select " +
            "class='form-control' name='" + fieldName + "'></select>";
    }

    function getLabel() {
        return "<label class='control-label'>" + label + "</label>";
    }

    this.getFieldByType = function (fieldType) {
        // const fieldType = obj.fieldTypeID;

        let field = this.fieldsModel[fieldType].getHtml();
        return "<div class='form-group'>" + field + "</div>";
    };

    this.getField = function () {
        this.buildField();
        return this.field;
    };
};

File: myapp/helpers/QuickField.py
class QuickField:
    name = ""
    type = ""
    id = ""
    field = ""
    label = ""
    classes = []
    maxlength = ""
    placeholder = ""
    style = ""

    def __init__(self, name, label="", classes=[], size=3, type="text", id="", maxlength=155,
                 multiple=False, placeholder="", style=""):
        self.name = str(name)
        self.placeholder = str(placeholder)
        self.style = str(style)
        self.id = str(id)
        self.type = str(type)
        self.label = str(label)
        self.classes = classes
        self.maxlength = maxlength
        self.size = str(size)
        if not label:
            self.label = name
        if not id:
            self.id = name

    def build_field(self):
        self.field = self.get_label() + self.get_input()

        return self.get_col()

    def get_col(self):
        return "<div class=\"col-md-" + self.size + "\">\n\t<div class=\"form-group\">\n" + str(
            self.field) + "\t</div>\n</div>"

    def get_input(self, type="text"):
        attrs = self.get_field_attrs()

        return "\t\t<input maxlength=\"" + str(self.maxlength) + "\" type=\"" + type + "\" " + attrs + ">\n"

    def get_select(self):
        attrs = self.get_field_attrs()
        options = self.get_select_options()

        return "\t\t<select " + attrs + ">" + options + "</select>\n"

    def get_textarea(self):
        attrs = self.get_field_attrs()

        return "\t\t<textarea " + attrs + "></textarea>\n"

    def get_select_options(self):
        return "<option ></option>"

    def get_field_attrs(self):
        is_required = False
        placeholder = ""
        if self.placeholder:
            placeholder = "placeholder=\"" + self.placeholder + "\" "

        style = ""
        if self.style:
            style = "style=\"" + self.style + "\" "

        return style + placeholder + "class=\"form-control " + " ".join(self.classes) + "\"" + (
            "required" if is_required else "") + \
               " id=\"" + self.id + "\" name=\"" + self.name + "\""

    def get_label(self):
        return "\t\t<label for=\"" + self.id + "\" class=\"control-label\">" + self.label + "</label>\n"

    def get(self):
        return self.build_field()

    def __str__(self):
        return self.get()


C = QuickField("NomePaciente", "Nome", style="x")
print(C)
